Orange/widgets/visualize/owdistributions.py

```diff
@@ -582,9 +588,28 @@ def send_report(self):
         self.report_caption(text)
 
     @classmethod
-    def migrate_context(cls, context, version):
-        if version < 2:
-            pass  # cannot convert index to variable
+    def migrate_context(cls, context, version, *, self=None, domain=None):
+        """
+        Migrate old indices into combos (`variable_idx`, `groupvar_idx`) to
+        variables (`var`, `group_var`). This cannot be done through usual
+        migrations because it requires a domain. The code is still put into
+        this method just so that anybody looking for it will find it here.
+        The patch is called after `openContext`
+        """
+        if version is None or version < 2:
+            if domain is not None:  # Called after openContext, not migrations
+                settings.migrate_idx_to_variable(
+                    self, "variable_idx", "var", self.varmodel,
+                    (var for var in chain(domain.variables, domain.metas)
+                     if var.is_primitive()))
+                settings.migrate_idx_to_variable(
+                    self, "groupvar_idx", "group_var", self.groupvarmodel,
+                    (var
+                     for var in chain([None], domain.variables, domain.metas)
+                     if var is None or var.is_primitive()))
```

Comment:
The code in the lines above has a bug or bugs. The widget does not behave correctly and a test fails. Now the widget has **Group by** set to **(None)** by default.

Reply:
True, but this is related to what we have to decide about.

It does migrate the settings. However, if we keep the obsolete settings just to make sure they are copied into the widget (and be visible to this method), they will always be present and they'll override the new settings. I thought `if version is None or version < 2:` would prevent this, but it obviously can't -- `version` is always `None` here. :)

I'll see how to finish this after we decide how to proceed with porting these settings. At this point, the widget can't know whether `groupvar_idx` is there from old settings or not.